frontend/src/pages/SignupPage/SignupPage.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+import axios, { AxiosError,HttpStatusCode } from 'axios';
+import { ToastContainer, toast } from 'react-toastify';
+import { AUTH_REGISTER_URL } from '../../constants/constants';
+import SignupFormData from '../../interfaces/SignupFormData';
+import isEmailValidCheck from '../../utils/isEmailValidCheck';
+
+import 'react-toastify/dist/ReactToastify.css';
+import './SignupPage.css'
+
+const SignupPage = () => {
+  const [signupFormData, setSignupFormData] = useState<SignupFormData>({
+    username: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const handleInputChange = (fieldName:string, value:string) => {
+    setSignupFormData({ ...signupFormData, [fieldName]: value });
+  };
+
+  const isFormDataValidCheck = () => {
+    const { username, email, password, confirmPassword } = signupFormData;
```

Comment:
```suggestion
    const { userName, email, password, confirmPassword } = signupFormData;
```

Tip: if you're using VSCode, highlight the variable name, click F2 to change the name once and across the file in one go.